pandas/core/indexes/base.py

```diff
@@ -1832,15 +1832,31 @@ def __contains__(self, key):
             return False
 
     _index_shared_docs['contains'] = """
-        return a boolean if this key is IN the index
+        Return a boolean if key is in the index.
+
+        This function returns True if key is in the index and
+        returns False if key is not in index.
 
         Parameters
         ----------
-        key : object
+        key : immutable object
```

Comment:
can remove `immutable`, it could be anything.

Reply:
can't be anything:
pd.CategoricalIndex([2000, 2001, 2012]).contains([1,2])
TypeError: unhashable type: 'list'